### ISSUE TITLE
serve-static redirect is ignored when express-static-gzip index is true and non-root mount is used

### ISSUE THREAD

**Issue by bdolgov** [tkoenig89/express-static-gzip#68]
Serve-static `redirect` option (which redirects requests without a trailing `/` to the ones with trailing `/`, if a directory is requested) does not work when express-static-gzip is used with a non-root mount point, and this mount point is requested:

```
let app = express();
app.use('/prefix', expressStaticGzip('/some/path');
app.listen(...);
```

On `GET /prefix`, serve-static returns a redirect to `/prefix/`,but express-static-gzip returns the contents of `/some/path/index.html`.

This happens because express rewrites both `/prefix` and `/prefix/` to `/` before calling express-static-gzip, and there is no way to distinguish whether the original request contained the trailing `/` from the rewritten URL. Serve-static inspects the original URL to solve the problem: [code](https://github.com/expressjs/serve-static/blob/87c5f092ed7cd23202f699b08c7f407c2b536df0/index.js#L90). Ideally, express-static-gzip should do the same.

**Comment by tkoenig89** [comment]
Thanks for providing a PR for this. I will try to release a new version soon.

**Comment by tkoenig89** [comment]
changes are released in version 2.2.0
